ostd/src/collections/rcu_xarray/cursor.rs

```diff
@@ -0,0 +1,413 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::ops::{Deref, DerefMut};
+
+use super::{
+    entry::{NodeEntryRef, XEntry},
+    mark::{NoneMark, XMark},
+    node::{Height, XNode},
+    XArray, SLOT_SIZE,
+};
+use crate::{
+    sync::non_null::{NonNullPtr, OwnedPtrRef},
+    task::atomic_mode::AsAtomicModeGuard,
+};
+
+/// A type representing the state of a [`Cursor`] or a [`CursorMut`]. Currently,
```

Comment:
Please read all the documentations yourselves and ensure the first sentence is a summary sentence. Don't include detailed explanation like "Currently, there are two possible states: [..]".

See https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components.